### ISSUE TITLE
file upload fails when using webrtc

### ISSUE THREAD

**Issue by paolor1965** [Ylianst/MeshCentral#6309]
**Describe the bug**
if I try to upload a file from the files menu on a Windows device regardless of the version, the copy fails and as long as upload is active a temporary file of 0 bytes is created on the remote device
the problem occurs with all versions after 1.1.24
file download works

**To Reproduce**
1. Go to 'any windows controlled device'
2. Click on 'files.'
3. I choose a file and start uploading
4. the progress bar does not start

**Expected behavior**
I expect the 150MB file to be copied to  chosen location
**Screenshots**
![immagine](https://github.com/user-attachments/assets/e8609320-bcec-466f-819f-c9d1fdbd0118)

**Server Software (please complete the following information):**
 - OS: AlmaLinux release 8.10 (Cerulean Leopard)
 - Virtualization: [proxmox guest]
 - Network: [LAN/WAN]
 - Version: [e.g. >=1.1.25]
 - Node: [v18.17.1]

**Client Device (please complete the following information):**
 - Device: desktop
 - OS: [windows 10 ltsc]
 - Network: [Remote over WAN]
 - Browser: [ firefox 128.0.2]
 - MeshCentralRouter Version: [if applicable]

**Remote Device (please complete the following information):**
 - Device: [desktop]
 - OS: [e.g. Windows 10 ltsc+ windows 11 23h2 ]
 - Network: [ Remote over WAN]
 - Current Core Version (if known): [**HINT**: Go to a device then `console` Tab then type `info`]
Current Core: Dec 9 2022, 1807787178
Agent Time: 2024-08-06 14:59:56.390+02:00.
User Rights: 0xffffffff.
Platform: win32.
Capabilities: 15.
Server URL: wss://meshcentral2.xxx.xxx.xxx.xxx:443/agent.ashx.
OS: Microsoft Windows 11 Pro - 23H2/22631.
**Additional context**

**Your config.json file**

**Comment by paolor1965** [comment]
yes thanks I didn't have time to sanitize it. I'll try right away

**Comment by si458** [comment]
great! so the must be a problem then with the webrtc!
ill change title and have a look when i get chance, as i know webrtc causes ALOT of problems for different things

**Comment by si458** [comment]
@paolor1965 can you share your config.json please so i can do some testing?

**Comment by paolor1965** [comment]
iao, eccoti il config.json privato di parti sensibili che sono sostiuite con delle xxx. Chiedo scusa see non è un esempio di pulizia

[config.json](https://github.com/user-attachments/files/16846210/config.json)

**Comment by si458** [comment]
I have actually been able to replicate the issue

The issue seems to be because we increased the bytes uploaded for normal file transfers, which speeded up normal file transfers.

But for some reason, the webrtc seems to be killing the meshagent and restarting it during a file transfer now with webrtc and no idea why?

So this isn't a simple fix, sadly but still looking into it

**Comment by paolor1965** [comment]
thanks for what you are doing, unfortunately I am not a programmer otherwise I would have helped

**Comment by si458** [comment]
@paolor1965 no it's OK!
The simple work around for the moment is set `webrtc: false` 
This disabled webrtc completely and relays everything through ur meshcentral
Not ideal but works 100% of the time!

**Comment by paolor1965** [comment]
I understand, in what other cases besides file transfer does disabling webrtc affect,

**Comment by si458** [comment]
Webrtc was used in the remote desktop and file transfer from what I've checked
It wasn't used the in terminal
But again disabling webrtc won't do any harm
It's just the connection goes through ur meshcentral server rather than going directly to the remote device if it can

It actually starts out by relaying though ur meshcentral server anyways when u click connect, it's just within the first 5 seconds both ur device and remote device then attempt to do webrtc and if that's possible, it then switches over to using that :)

**Comment by paolor1965** [comment]
thanks everything clear

**Comment by si458** [comment]
ok so this should be fixed in 1.1.29, BUT ive noticed for some reason,
depending on the remote machine, it can disconnect and reconnect during the transfer?
this is a seperate issue tho

**Comment by Henrique-Batista** [comment]
@si458 I think the problem is related to Websocket compression. I noticed that in both my configuration and @paolor1965 's, WebRTC is disabled, but agentWsCompression and WsCompression are enabled. I tried disabling WsCompression and leaving WebRTC enabled (actually, it doesn't matter if it's enabled or disabled) and the problem didn't occur. I also checked the web server request tracking, and when I enable WSCompression, the web request doesn't seem to reach the server; it only works when WSCompression is disabled. I don't know about paolor's environment, but my environment has some particular settings that might influence this issue. So if you can't reproduce the problem, let me know and I'll create another issue to provide the information.

**Comment by si458** [comment]
@Henrique-Batista I will try to have a look when I get a chance as i dont use any compression at all for anything and i have issues from time to time with webrtc whcih is why i tell people to NOT use it!

Im currently ill as hell, full of flu/cold, snotting everywhere and making it worse, I've now got a serious tooth ake/pain, so that is another expensive emergency nhs dentist appointment to come 😞

**Comment by Henrique-Batista** [comment]
Hello @si458 , I hope you are better. I did some tests in my environment and with WebRTC active I can't access the machines because it crashes milliseconds after starting and the Files directories don't change without disconnecting and connecting again. I believe this is due to the configuration I have here, but I've also had other problems because of WebRTC.
